accelerator/iri_proxy_apis.c

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright (C) 2018-2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include "iri_proxy_apis.h"
+#include "utils/handles/lock.h"
+
+#define IRI_PROXY_APIS_LOGGER "iri_proxy_apis"
+
+static logger_id_t logger_id;
+lock_handle_t cjson_lock;
+
+void iri_proxy_apis_logger_init() { logger_id = logger_helper_enable(IRI_PROXY_APIS_LOGGER, LOGGER_DEBUG, true); }
+
+int iri_proxy_apis_logger_release() {
+  logger_helper_release(logger_id);
+  if (logger_helper_destroy() != RC_OK) {
+    ta_log_critical("Destroying logger failed %s.\n", IRI_PROXY_APIS_LOGGER);
+    return EXIT_FAILURE;
+  }
+
+  return 0;
+}
+
+status_t iri_proxy_apis_lock_init() {
+  if (lock_handle_init(&cjson_lock)) {
+    return SC_CONF_LOCK_INIT;
+  }
+  return SC_OK;
+}
+
+status_t iri_proxy_apis_lock_destroy() {
+  if (lock_handle_destroy(&cjson_lock)) {
+    return SC_CONF_LOCK_DESTROY;
+  }
+  return SC_OK;
+}
+
+status_t res_to_json_result(char_buffer_t* res_buff, char** json_result) {
```

Comment:
May you rename this function into `char_buffer_to_str`?
I think it would be more intuitively.

Reply:
Copy that!